docs: everest mount page refresh (#9601)
treeverse/lakeFS: docs/src/reference/mount.md

!!! info
    Available in **lakeFS Cloud** and **lakeFS Enterprise**

Everest is a complementary binary to lakeFS that allows you to virtually mount a remote lakeFS repository onto a local directory or within a Kubernetes environment. Once mounted, you can access data as if it resides on your local filesystem, using any tool, library, or framework.

<iframe width="560" height="315" src="https://www.youtube.com/embed/BgKuoa8LAaU" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

## Use Cases

- **Simplified Data Loading**: Use your existing tools to read and write files directly from the filesystem with no need for custom data loaders or SDKs.
- **Seamless Scalability**: Scale from a few local files to billions without changing your tools or workflow. Use the same code from experimentation to production.

> - **Enhanced Performance**: Everest supports billions of files and offers fast, lazy data fetching, making it ideal for optimizing GPU utilization and other performance-sensitive tasks.

Reviewer: The use case title sounds like a technical benefit of mount. But there is a use case for performant data loading which I believe should be highlighted instead. WDYT?

---

## Getting Started

This guide will walk you through setting up and using Everest to mount a lakeFS repository on your local machine.

!!! tip "New to Everest?"
    After completing this getting started guide, we recommend reading the [Core Concepts](#core-concepts) section to understand caching, consistency, and performance characteristics.

> ### 1. Prerequisites

Reviewer: nit; I would remove the numbers, it reduces clarity

    export EVEREST_LAKEFS_CREDENTIALS_PROVIDER_AWS_IAM_CLIENT_LOG_PRE_SIGNING_REQUEST=true
    ```

### 3. Your First Mount (Read-Only)

Let's mount a prefix from a lakeFS repository to a local directory. In read-only mode, Everest mounts a specific commit ID. If you provide a branch name, it will resolve to the HEAD commit at the time of mounting.

1.  **Mount the repository:**
    This command mounts the `datasets/pets/` prefix from the `main` branch of the `image-repo` repository into a new local directory named `./pets`.

    ```bash
    everest mount "lakefs://image-repo/main/datasets/pets/" "./pets"
    ```

2.  **Explore the data:**
    You can now use standard filesystem commands to interact with your data. Files are downloaded lazily only when you access their content.

    ```bash
    # List files - this only fetches metadata
    ls -l "./pets/dogs/"

    # Find files
    find ./pets -name "*.small.jpg"

    # Open a file - this triggers a download
    open -a Preview "./pets/dogs/golden_retrievers/cute.jpg"
    ```

3.  **Unmount the directory:**
    When you are finished, unmount the directory.

    ```bash
    everest umount "./pets"
    ```

---

## Core Concepts

> Before diving into data operations, it's important to understand how Everest manages performance, consistency, and caching. These concepts apply to both local and Kubernetes deployments.

Reviewer: ```suggestion
This section will help you understand how Everest manages performance, consistency, and caching in both local and Kubernetes deployments.
```

### Cache Behavior

> Everest uses a local cache to improve performance when accessing files from lakeFS. Understanding how the cache works will help you optimize performance for your specific use case.

Reviewer: I think that the guide should give explicit guidelines for how to optimize performance. This line makes the reader work harder to try and keep up and understand how to to it

Author: This section was added as part of questions to understand better how the cache works.

Your comment is for us adding another section about how to improve performance and keeping this section to understand how the cache works?

> #### How Caching Works

Reviewer: Can we keep all heading 4 out of the toc?

**Key points:**

-   Each new mount creates a fresh cache directory.
-   By default cache location is managed by Everest and cleaned up automatically.
-   The cache is ephemeral and does not persist between mount sessions. Unless you specify the cache directory.

#### Persistent Cache

To reuse cache data across multiple mount sessions, you can specify a custom cache directory using the `--cache-dir` flag:

```bash
everest mount lakefs://image-repo/main/datasets/ ./datasets --cache-dir ~/.everest-cache
```

**Benefits of persistent cache:**

-   Faster startup times when remounting the same data.
-   Reduced bandwidth usage by reusing previously downloaded files.
-   Useful for iterative workflows where you repeatedly mount and unmount the same repository.

#### Cache Management

Everest manages cached data based on the commit ID of the mounted reference:

-   **Commit-Based Caching**: Each commit ID has its own cache namespace. This ensures that cached data always corresponds to the correct version of your files.

> -   **Cache Invalidation on Commit**: When you commit changes in write mode using `everest commit`, the mount point's source commit ID is updated to the new HEAD of the branch. As a result, the cache associated with the old commit ID is no longer used, and new data will be cached under the new commit ID.

Author: sure - each release we will need to update the documentation to reflect the user whats running on their machine.

!!! tip "Optimizing Cache Size"
    Set `--cache-size` to match the amount of data you plan to read or write. A larger cache reduces the need to evict and re-fetch files, improving performance for workloads that access many files.

### Consistency & Data Behavior

> Understanding how Everest handles data consistency is crucial for working effectively with mounted lakeFS repositories.

Reviewer: I would remove this line

#### File System Consistency

Everest mount provides **strong read-after-write consistency** within a single mount point. Once a write operation completes, the data is guaranteed to be available for subsequent read operations on that same mount.

#### lakeFS Consistency

Local changes are reflected in lakeFS only after they are **committed** using the `everest commit` command. Until then:

-   Changes are only visible within your local mount point
-   Other users or mounts will not see your changes
-   If two users mount the same branch, they will not see each other's changes until those changes are committed

#### Sync Operation

When you run `everest diff` or `everest commit`, Everest performs a **sync operation** that uploads all local changes to a temporary location in lakeFS for processing. This ensures your changes are safely transferred before being committed to the branch.

See the [Write-Mode Operations](#write-mode-operations) section for more details on working with writable mounts.

### Performance Considerations

> Everest is optimized for high-performance data access, but understanding these characteristics will help you get the best results:

Reviewer: ```suggestion
Everest achieves high-performance data access through:
```

-   [Status and Limitations](#status-and-limitations) - Supported platforms and current limitations
-   [Prerequisites](#1-prerequisites) - Requirements for deploying the CSI driver
-   [Deploy the CSI Driver](#2-deploy-the-csi-driver) - Installation instructions using Helm
-   [Use in Pods](#3-use-in-pods) - How to mount lakeFS URIs in your Kubernetes workloads
-   [Troubleshooting](#4-troubleshooting) - Common issues and debugging steps

### How it Works

The CSI driver, installed in your cluster, orchestrates mount operations on each Kubernetes node. It does not execute `mount` commands directly. Instead, it communicates via a Unix socket with a `systemd` service running on the host. This service is responsible for executing the `everest mount` and `umount` commands, making lakeFS URIs available to Pods as persistent volumes.

### Status and Limitations

-   **Tested OS:** BottleRocket-OS, Amazon Linux 2, RHEL 8.
-   **Kubernetes:** Version `>=1.23.0`.
-   **Provisioning:** Static provisioning only.
-   **Access Modes:** `ReadOnlyMany` is supported.
-   **Security Context:** Setting Pod `securityContext` (e.g., `runAsUser`) is not currently supported.

> ### 1. Prerequisites

Author: Prerequisites at this level is part of the CSI driver
